HeterogeneousCore/SonicTriton/python/customize.py

```diff
@@ -0,0 +1,174 @@
+        process.TritonService.fallback.device = options.device
+        if len(options.fallbackName)>0:
+            process.TritonService.fallback.instanceBaseName = options.fallbackName
+        if len(options.address)>0:
+            process.TritonService.servers.append(
+                cms.PSet(
+                    name = cms.untracked.string(options.serverName),
+                    address = cms.untracked.string(options.address),
+                    port = cms.untracked.uint32(options.port),
+                    useSsl = cms.untracked.bool(options.ssl),
+                    rootCertificates = cms.untracked.string(""),
+                    privateKey = cms.untracked.string(""),
+                    certificateChain = cms.untracked.string(""),
+                )
+            )
+
+    if applyToModules:
+        process = configureModules(process, **getClientOptions(options))
+
+    return process
+
+def getClientOptions(options):
+    return dict(
+        compression = cms.untracked.string(options.compression),
+        useSharedMemory = cms.untracked.bool(not options.noShm),
+        timeout = cms.untracked.uint32(options.timeout),
+        timeoutUnit = cms.untracked.string(options.timeoutUnit),
+        allowedTries = cms.untracked.uint32(options.tries),
+    )
+
+def applyClientOptions(client, options):
+    return configureClient(client, **getClientOptions(options))
+
+def configureModules(process, modules=None, returnConfigured=False, **kwargs):
+    if modules is None:
+        modules = {}
+        modules.update(process._Process__producers)
+        modules.update(process._Process__filters)
+        modules.update(process._Process__analyzers)
+    configured = []
+    for pname,producer in modules.items():
+        if hasattr(producer,'Client'):
+            producer.Client = configureClient(producer.Client, **kwargs)
+            configured.append(pname)
+    if returnConfigured:
+        return process, configured
+    else:
+        return process
+
+def configureClient(client, **kwargs):
+    client.update_(kwargs)
+    return client
+
+def configureLogging(process, client=False, server=False, service=False, discovery=False):
+    if not any([client, server, service, discovery]):
+        return
+
+    keepMsgs = []
+    if discovery:
+        keepMsgs.append('TritonDiscovery')
+    if client:
+        keepMsgs.append('TritonClient')
+    if service:
+        keepMsgs.append('TritonService')
+
+    if hasattr(process,'TritonService'):
+        process.TritonService.verbose = service or discovery
+        process.TritonService.fallback.verbose = server
+    if client:
+        process, configured = configureModules(process, returnConfigured=True, verbose = True)
+        for module in configured:
+            keepMsgs.extend([module, module+':TritonClient'])
+
+    if not hasattr(process,'MessageLogger'):
+        process.load('FWCore/MessageService/MessageLogger_cfi')
```

Comment:
Just to note that this piece is needed only if the user has explicitly deleted `process.MessageLogger`. `MessageLogger` has been part of `cms.Process` for quite some time already.

If kept, while this works, I'd prefer
```suggestion
    if not hasattr(process,'MessageLogger'):
        process.load('FWCore.MessageService.MessageLogger_cfi')
```

Reply:
Ah, I actually did not know it was included in the initialization of `cms.Process`. Then I think this condition is not necessary.